backend/src/sql/list-dreps.sql

```diff
@@ -29,22 +29,88 @@ DRepStatus AS (
   FROM
     drep_hash dh
     LEFT JOIN (
-      SELECT
+      -- Replace ROW_NUMBER with DISTINCT ON
+      SELECT DISTINCT ON (dr.drep_hash_id)
         dr.id,
         dr.drep_hash_id,
-        dr.deposit,
-        ROW_NUMBER() OVER (PARTITION BY dr.drep_hash_id ORDER BY dr.tx_id DESC) AS rn
+        dr.deposit
       FROM
         drep_registration dr
       WHERE
         dr.deposit IS NOT NULL
+      ORDER BY dr.drep_hash_id, dr.tx_id DESC
     ) AS dr_deposit ON dr_deposit.drep_hash_id = dh.id
-    AND dr_deposit.rn = 1
     LEFT JOIN tx ON tx.id = dr_deposit.id
     LEFT JOIN block ON block.id = tx.block_id
     LEFT JOIN block AS block_first_register ON block_first_register.id = tx.block_id
     CROSS JOIN DRepActivity
   GROUP BY dh.id, dr_deposit.deposit, DRepActivity.epoch_no, DRepActivity.drep_activity
+),
+LatestDRepRegistrations AS (
+  SELECT DISTINCT ON (dr.drep_hash_id)
+    dr.id,
+    dr.drep_hash_id,
+    dr.deposit,
+    dr.voting_anchor_id,
+    dr.tx_id,
+    tx.hash AS tx_hash
+  FROM
+    drep_registration dr
+    JOIN tx ON tx.id = dr.tx_id
+  ORDER BY dr.drep_hash_id, dr.tx_id DESC
+),
+NonDeregisterVotingAnchor AS (
+  -- Replace ROW_NUMBER with DISTINCT ON
+  SELECT DISTINCT ON (dr.drep_hash_id)
+    dr.id,
+    dr.drep_hash_id,
+    dr.voting_anchor_id,
+    tx.hash AS tx_hash
+  FROM
+    drep_registration dr
+    JOIN tx ON tx.id = dr.tx_id
+  WHERE 
+    dr.deposit IS NOT NULL
+    AND dr.deposit >= 0
+  ORDER BY dr.drep_hash_id, dr.tx_id DESC
+),
+SecondNewestRegistration AS (
+  -- Replace ROW_NUMBER with a more complex query to get second newest
+  SELECT DISTINCT ON (dr.drep_hash_id)
+    dr.id,
+    dr.drep_hash_id,
+    dr.voting_anchor_id
+  FROM
+    drep_registration dr
+  WHERE
+    dr.tx_id < (
+      SELECT MAX(tx_id) 
+      FROM drep_registration dr2 
+      WHERE dr2.drep_hash_id = dr.drep_hash_id
+    )
+  ORDER BY dr.drep_hash_id, dr.tx_id DESC
+),
+NewestRegister AS (
+  -- Replace ROW_NUMBER with DISTINCT ON
+  SELECT DISTINCT ON (dr.drep_hash_id)
+    block.time,
+    dr.drep_hash_id
+  FROM
+    drep_registration dr
+    JOIN tx ON tx.id = dr.tx_id
+    JOIN block ON block.id = tx.block_id
+  WHERE
+    NOT (dr.deposit < 0)
+  ORDER BY dr.drep_hash_id, dr.tx_id DESC
+),
+FirstRegister AS (
+  -- Replace ROW_NUMBER with DISTINCT ON
+  SELECT DISTINCT ON (dr.drep_hash_id)
+    dr.tx_id,
+    dr.drep_hash_id
+  FROM
+    drep_registration dr
+  ORDER BY dr.drep_hash_id, dr.tx_id ASC
```

Comment:
If the intention of this cte is to just take the block time, you can join the block inside of it instead of joining on the final Select